android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -65,6 +66,49 @@ public TiListView(ListViewProxy proxy)
 		this.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
 		this.recyclerView.setFocusableInTouchMode(false);
 
+		// Add listener to fire scroll events.
+		this.recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener()
+		{
+			@Override
+			public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState)
+			{
+				super.onScrollStateChanged(recyclerView, newState);
+
+				if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+					proxy.fireSyncEvent(TiC.EVENT_SCROLLEND, generateScrollPayload());
+				}
+			}
+
+			@Override
+			public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy)
+			{
+				super.onScrolled(recyclerView, dx, dy);
+
+				proxy.fireSyncEvent(TiC.EVENT_SCROLLSTART, generateScrollPayload());
+			}
+		});
+		this.recyclerView.setOnFlingListener(new RecyclerView.OnFlingListener()
+		{
+			@Override
+			public boolean onFling(int velocityX, int velocityY)
+			{
+				final KrollDict payload = new KrollDict();
+
+				// Determine scroll direction.
+				if (velocityY > 0) {
+					payload.put(TiC.PROPERTY_DIRECTION, "down");
+				} else if (velocityY < 0) {
+					payload.put(TiC.PROPERTY_DIRECTION, "up");
+				}
+
+				// Set scroll velocity.
+				payload.put(TiC.EVENT_PROPERTY_VELOCITY, velocityY);
+
+				proxy.fireSyncEvent(TiC.EVENT_SCROLLING, payload);
+				return true;
+			}
```

Comment:
Does doing this via `onFling()` cause the "scrolling" event to fire as you drag your finger? Because that's how it used to work.

And in Titanium 9.1.0, we did dampen the event firing based on the device's "touch slop" setting. This prevents a lot of up/down events from being spammed when you touch and hold on the ListView on a sensitive screen. You have to travel a minimum distance before firing the event.
https://github.com/appcelerator/titanium_mobile/pull/11680

Reply:
It only fires after lifting from a drag, where it can then calculate the velocity/fling of the gesture

Reply:
I re-tested this on Android with an older version of the Titanium SDK. The old Android behavior was that the "scrolling" event was only fired when you change vertical direction. You do not have to release your finger. Everytime you drag up/down with your finger on the screen you'll get this event. And the event is not spammed as you scroll in the same direction.

However, the iOS behavior is the "scrolling" event is only fired after you release your finger from the screen. So if you fling it and release your finger, you will get a scrolling event indication the direction it as flinged in. If you drag your finger, stop scrolling (don't fling), and then release your finger then you'll get a "scrolling" event with an "undefined" direction. This matches what we document below.
https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.ListView-event-scrolling

I personally prefer Android's old behavior compared to iOS because at least you can use it hide/show the top toolbar like how Google does it... but whatever.